### DISCUSSION TITLE
Problem of TLS connection of the tasmota device with its server

### DISCUSSION THREAD

**Discussion by songhellobulter** [arendst/Tasmota#22583] [Q&A]
Our server has SSL.com certificate like in the attached.

We generated the certificate data and saved them in local_ca_descriptor.h, local_ca_data.h 
and #define INCLUDE_LOCAL_CERT in the user_config_override.h

Now we have TLS connection error 296.
![UAT server SSL](https://github.com/user-attachments/assets/4f044610-fae7-4afa-ab7f-da7526339da0)

Have we missed anything. Could anyone please suggest? Thank you very much.

**Comment by s-hadinger** [comment]
Error 296 is `SSL3_ALERT_HANDSHAKE_FAILURE`

This makes me believe that you server is expecting a certificate based authentication from the client, which we have very limited support.

**Reply by songhellobulter** [reply]
Thank you for s-hadinger for your answer.
The server works fine with our other product with a same ESP32 chip which includes the same certificate data. Should it work OK with the tasmota one? 
From the server certificate, it is RSA and supports ECDHE_RSA_WITH_AES_128_GCM_SHA256. What else should we need?

**Comment by s-hadinger** [comment]
By the way, if you want to test with bearssl on a laptop, compile bearssl and try:

`./brssl client <ip_or_domain>:<port> -cs ECDHE_RSA_WITH_AES_128_GCM_SHA256`
